Mathlib/LinearAlgebra/Finsupp/LinearCombination.lean

```diff
@@ -411,6 +411,11 @@ theorem mem_span_range_iff_exists_fun :
   simp only [Finsupp.mem_span_range_iff_exists_finsupp, Finsupp.equivFunOnFinite_apply]
   exact exists_congr fun c => Eq.congr_left <| Finsupp.sum_fintype _ _ fun i => zero_smul _ _
 
+omit [Fintype α] in
+theorem mem_span_image_iff_exists_fun {s : Set α} [Fintype s] :
+    x ∈ span R (v '' s) ↔ ∃ c : s → R, ∑ i, c i • v i = x := by
+  rw [← mem_span_range_iff_exists_fun, image_eq_range]
```

Comment:
Hmm, I feel like the forward direction is what will be used most. But ideally also in cases where `s` isn't necessarily finite. So could you please modify the statement so that it says there is a `t : Finset \a` with `t \subset s` such that yaddayadda?

Reply:
OK I've added a version with the sort of statement I think you had in mind (and which does not assume finiteness). I have retained what I had though: I claim that we will benefit from having both. As evidence I highlight that my use case is much more convenient with the original version.

Reply:
(I do however agree it is worth having the version you describe.)